mobile/lib/modules/backup/background_service/background.service.dart

```diff
@@ -50,6 +50,11 @@ class BackgroundService {
       _Throttle(_updateProgress, notifyInterval);
   late final _Throttle _throttledDetailNotify =
       _Throttle(_updateDetailProgress, notifyInterval);
+  Completer<bool> _hasAccessCompleter = Completer();
```

Comment:
I've never heard of `Completer` before, does it work similar to a `Promise` in Javascript or `Result` type in Rust?

Reply:
It's somewhat related to `Promise`. A `Completer` is the low-level building block in dart to create `Future`s that can be `await`ed from callback-based (or similar) code.